commit 307c24681421b316add18f7c1dfa85c123f836c0
Author: Sergei Trofimovich <slyfox@gentoo.org>
Date:   Tue Nov 26 00:39:51 2013 +0300

    src/Nxt/Graphics.hs: fix against opengl-2.9
    
    > src/Nxt/Graphics.hs:91:5:
    >     No instance for (TwoDimensionalTextureTarget (Maybe a0))
    >       arising from a use of `texImage2D'
    
    Reported by kini.
    
    Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>

diff --git a/src/Nxt/Graphics.hs b/src/Nxt/Graphics.hs
index 981d6df..384fbd8 100644
--- a/src/Nxt/Graphics.hs
+++ b/src/Nxt/Graphics.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 module Nxt.Graphics
     (begin,
      end,
@@ -88,7 +89,13 @@ loadTexture textureFilePath = do
     surfacePixels <- surfaceGetPixels surface
 
     let pixelData = PixelData RGBA UnsignedByte surfacePixels
-    texImage2D Nothing NoProxy 0 RGBA' surfaceSize 0 pixelData
+    texImage2D
+#if MIN_VERSION_OpenGL(2,9,0)
+          Texture2D
+#else
+          Nothing
+#endif
+          NoProxy 0 RGBA' surfaceSize 0 pixelData
 
     freeSurface surface
 
